cpp/controller/controller.cpp

```diff
@@ -274,3 +321,10 @@ void Controller::onTimerStopRequested()
     stop();
 }
 
+void Controller::onCyclesModeChanged(bool cyclesMode)
+{
+    m_cyclesController.resetCurrentInterval();
+    if (cyclesMode && m_state > State::Off) {
```

Comment:
I think `m_state == State::Off || m_state == Paused` is less error prone

Reply:
I agree. But (what's kind of funny ;) ) you've made an error in your less erron prone solution.
It should be `Working || Paused`.